pytorch_lightning/utilities/cli.py

```diff
@@ -293,10 +375,32 @@ def __init__(
             self.fit()
             self.after_fit()
 
+    @property
+    def registered_optimizers(self) -> Tuple[Type[Optimizer]]:
+        return tuple(OPTIMIZER_REGISTRY.values())
+
+    @property
+    def registered_lr_schedulers(self) -> Tuple[LRSchedulerType]:
+        return tuple(LR_SCHEDULER_REGISTRY.values())
+
     def init_parser(self, **kwargs: Any) -> LightningArgumentParser:
         """Method that instantiates the argument parser."""
         return LightningArgumentParser(**kwargs)
 
+    def _sanitize_argv(self) -> None:
+        """This function is used to replace space within `sys.argv` with its equal sign counter-part."""
+        args = [idx for idx, v in enumerate(sys.argv) if v.startswith("-")]
+        if not args:
+            return
+        start_index = args[0]
+        argv = []
+        for v in sys.argv[start_index:]:
+            if v.startswith("-"):
+                argv.append(v)
+            else:
+                argv[-1] += "=" + v
+        sys.argv = sys.argv[:start_index] + argv
```

Comment:
It could be the normal case in LightningCLI that there is only one value for each argument. But it could be possible that someone adds an argument for example as `parser.add_argument('--myarg', nargs='+', type=str)`. A valid input to this could be `--myarg a b c`. This would become `--myarg=a=b=c` which is wrong. Also if an argument accepts a possibly negative number (value starting with "-") it does not get sanitized. These could be consider acceptable downsides. But still, is there a need to sanitize all args? If this is only to simplify `link_optimizers_and_lr_schedulers` it could be only for `--optimizer` and `--lr_scheduler`.

Reply:
The issue is that users could register their own nested key and those won't be optimizer or lr_scheduler

Reply:
Great point, resolved it.